sdk/logs/src/main/java/io/opentelemetry/sdk/logs/internal/SdkEventBuilder.java

```diff
@@ -35,6 +35,13 @@ class SdkEventBuilder implements EventBuilder {
     this.eventName = eventName;
   }
 
+  @Override
+  public <T> EventBuilder setPayload(AnyValue<T> payload) {
+    this.payload.clear();
+    ((ExtendedLogRecordBuilder) logRecordBuilder).setBody(payload);
```

Comment:
This might be the grossest part of this. It's also a little weird of an API because the builder could get reused and it's a bit of a chimera -- handling both simple types and complex types. Like, a confused user might call `put(), put(), put()` then `setPayload()` then `put(), put(), put()` again. 

Not sure why you'd ever want to or why one might think that should work, but it's _possible_. Just wanted to point it out.

Reply:
> handling both simple types and complex types. Like, a confused user might call put(), put(), put() then setPayload() then put(), put(), put() again.

Yeah in my head we'd solve this by having a nullable reference to the user set AnyValue payload. And then down in `emit`, if explicit payload was set, it takes priority over any calls to `put`. Something like:
```
@Nullable private AnyValue<?> explicitPayload;

...

public EventBuilder setPayload(AnyValue<?> payload) {
  this.explicitPayload = payloadl;
  return this;
}

...

@Override
public void emit() {
  if (explicitPayload != null) {
     ((ExtendedLogRecordBuilder) logRecordBuilder).setBody(explicitPayload);
  } else if (!payload.isEmpty()) {
    ((ExtendedLogRecordBuilder) logRecordBuilder).setBody(AnyValue.of(payload));
  }
  if (!hasTimestamp) {
    logRecordBuilder.setTimestamp(clock.now(), TimeUnit.NANOSECONDS);
  }
  logRecordBuilder.setAttribute(EVENT_NAME, eventName);
  logRecordBuilder.emit();
}
```

Potential to cause confusion to users, but can mitigate with javadoc explaining the behavior. You might say that this might be cause to remove / change the `put(String, ?)` methods, but I think that those are absolutely crucial from an ergonomics perspective.

Reply:
Yeah that's cool, the null was another approach that occurred to me, and I think the usability is fine and maybe slightly cleaner impl. I agree that those `put()` calls cannot go away!